### DISCUSSION TITLE
Add detailed warning message when resources is not included in the self-contained html (when exporting reveal.js presentation)

### DISCUSSION THREAD

**Discussion by ttzytt** [quarto-dev/quarto-cli#7775] [Q&A]
### Description

```qmd
---
format: 
    revealjs:
        self-contained: true
        embed-resources: true
        self-contained-math: true
---

## 

![](./img/quarto_logo.png)


## 

![](/img/quarto_logo.png)


```

For the second slide, the local resource `quarto_logo.png` will not be properly embedded into the self-contained HTML file: 

```html
<img data-src="/img/quarto_logo.png" class="r-stretch"><div class="footer footer-default">
```

I'm guessing that this is because the syntax of missing the dot before `/img` is wrong. However, this worked perfectly fine in the slide show if the HTML file was not moved from the original directory. So it seems like this syntax is fine for browsers. 

The current warning message is:

```
[WARNING] Could not fetch resource /img/quarto_logo.png
```

I think it is fine if Quarto has a different syntax than the browsers, but it will be helpful to add a reminder whenever it encounters a valid path with the dot missing (as I have spent a lot of time to figure out that it is the dot making the self-contained mode not working).

**Comment by mcanouil** [comment]
This is intended behaviour: `/` means root from the project while `./` means relative to the file calling the image, thus the message is correct.

Here I do not believe there is a need to another message, as if the resource is not found it obviously cannot be embedded, i.e., you don't have ˋimg/quarto_logo.png` at the root of the project, what details are you missing exactly beside your path is not correct in Quarto context?

Note that `self-contained` is a deprecated option (which is now an alias to `embed-resources`) as stated in the documentation and this is the case since September 2022. Please use only `embed-resources` as described everywhere in the documentation.

**Reply by ttzytt** [reply]
Thank you for the quick response!

This might sound stupid, but I just wonder what the "project path" is. Currently, I'm using Quarto inside VScode with the plugin, is the "project path" the path of the workspace you opened?

**Reply by mcanouil** [reply]
I don't know what you put into "project path".
A Quarto project is defined by its `_quarto.yml` (even if empty).
In theory, the Quarto extension for VSCode will assume the root of the workspace is the root of the project / working directory as with basically any projects (not only Quarto).
